public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilder.tsx

```diff
@@ -81,7 +81,11 @@ export const LokiQueryBuilder = React.memo<Props>(
 
       let values;
       const labelsToConsider = query.labels.filter((x) => x !== forLabel);
-      if (labelsToConsider.length === 0) {
+      // If we have no equality/regex operation with .*, we can't fetch series as it will throw an error, so we fetch label values
+      const hasEqualityOperation = labelsToConsider.find(
+        (filter) => filter.op === '=' || (filter.op === '=~' && filter.value !== '.*')
```

Comment:
I think what we really need is something like

```suggestion
        (filter) => filter.op === '=' || (filter.op === '=~' && new RegExp(filter.value).test("") === false)
```

to test for all variations of matchers that can match an empty string. That's what Loki also does in https://github.com/grafana/loki/blob/main/pkg/util/matchers.go#L16. The `matcher.Matches("")` is the equivalent of `new RegExp(filter.value).test("")`.